app/lib/api/file_system/file_system_dav.dart

```diff
@@ -199,11 +199,20 @@ class DavRemoteDocumentFileSystem extends DocumentRemoteSystem {
   @override
   Future<void> updateFile(String path, List<int> data,
       {bool forceSync = false}) async {
-    // Create directory if not exists
-    final directoryPath = p.dirname(path);
-    final directory = Directory(directoryPath);
-    if (!await directory.exists()) {
-      await directory.create(recursive: true);
+    // Cache check
+    if (!forceSync && remote.hasDocumentCached(path)) {
+      cacheContent(path, data);
+      return;
+    }
+
+    // Create directory if not exits
+    var last = path.lastIndexOf('/');
```

Comment:
Hmm i liked the p.dirname since it does handle things like `/directory/subdirectory/`.

Reply:
I didn't know about this dirname function before. Maybe we should add more of these useful path package functions in butterfly to improve error handling like trailing slashes?

Reply:
I can try, but I would like to solve the rest first. because I'm getting lost with all the comments 😂

Reply:
Sure but this and the return line removal are just the changes I would suggest, otherwise it loosk good!